my_list.rb

```diff
@@ -0,0 +1,18 @@
+require_relative('my_enumerable')
+class MyList
+  include MyEnumerable
+  def initialize(list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each { |a| block.call a }
+  end
+end
+
+list = MyList.new([1, 2, 3, 4])
```

Comment:
- [ ] the user should be able to add a list of numbers without it being an array already like the below example:
`list = MyList.new(1, 2, 3, 4)`
when running this code in the current application the user receives an error.
you can fix this in the initialize method by doing something similar to below:
```
    def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

Reply:
The array issue is fixed, thanks!